autometa/common/external/samtools.py

```diff
@@ -51,24 +51,29 @@ def sort(sam, bam, nproc=mp.cpu_count()):
     
     """
 
-    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {bam}'
+    samtools_out_dir = os.path.dirname(os.path.abspath(bam))
+    outfile_tail = os.path.basename(bam)
+    tempdir = tempfile.mkdtemp(suffix = None, prefix = 'samtools', dir = samtools_out_dir)
+    samtools_outfile_tail = os.path.join(samtools_out_dir, outfile_tail)
+    temp_outfile_tail = os.path.join(tempdir, outfile_tail)
+    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_outfile_tail}'
```

Comment:
There are currently no checks for the `nproc` parameter to be an integer greater than or equal to one. If None is provided, does this result in an error? 0?